ui/app/helpers/utils/i18n-helper.js

```diff
@@ -39,13 +40,41 @@ export const getMessage = (localeCode, localeMessages, key, substitutions) => {
   }
   const entry = localeMessages[key]
   let phrase = entry.message
+
+  const requiresWrappers = phrase.match(/\{\{.+?\}\}/)
+
+  const hasSubstitutions = Boolean(substitutions && substitutions.length)
+  const hasReactSubstitutions = hasSubstitutions && substitutions.some((element) => typeof element === 'function' || typeof element === 'object')
+
   // perform substitutions
-  if (substitutions && substitutions.length) {
+  if (hasReactSubstitutions || requiresWrappers) {
+    const parts = phrase.split(/(\$\d|\{\{.+?\}\})/g)
+
+    const wrappedAndSubStitutedParts = parts.map((part) => {
+      if (part.match(/\{\{.+?\}\}/)) {
+
+        const wrapper = reactWrappers.shift()
+        return React.cloneElement(
+          wrapper,
+          wrapper.props,
+          [ part.match(/[^{}]+/) ]
+        )
+      } else if (part.match(/\$\d/)) {
+        return substitutions.shift()
+      } else {
+        return part
+      }
+    })
+
+    phrase = React.createElement('span', null, ...wrappedAndSubStitutedParts)
```

Comment:
Why is `createElement` being used here instead of JSX?

Reply:
Changed to jsx in 247b12834